File: ios/ExpoRoomPlanViewModule.swift
import ExpoModulesCore
import UIKit
import RoomPlan

@available(iOS 17.0, *)
public class ExpoRoomPlanViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoRoomPlanView")

    // Register a React Native view that embeds RoomCaptureView
    View(RoomPlanCaptureUIView.self) {
      Events("onStatus", "onExported", "onPreview")

      // Props to control flow
      Prop("scanName") { (view, value: String?) in
        view.scanName = value
      }
      Prop("exportType") { (view, value: String?) in
        view.exportType = value
      }
      Prop("sendFileLoc") { (view, value: Bool?) in
        view.sendFileLoc = value ?? false
      }
      Prop("exportOnFinish") { (view, value: Bool?) in
        view.exportOnFinish = value ?? true
      }
      Prop("running") { (view, value: Bool?) in
        view.setRunning(value ?? false)
      }
      // Bump this number to trigger an export on demand
      Prop("exportTrigger") { (view, value: Double?) in
        view.handleExportTrigger(value)
      }
      // Stop capture and show RoomPlan preview UI
      Prop("finishTrigger") { (view, value: Double?) in
        view.handleFinishTrigger(value)
      }
      // Continue scanning and add another room to the set
      Prop("addAnotherTrigger") { (view, value: Double?) in
        view.handleAddAnotherTrigger(value)
      }
    }
  }
}

File: android/src/main/java/expo/modules/roomplan/ExpoRoomPlanViewModule.kt
package expo.modules.roomplan

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoRoomPlanViewModule : Module() {
  override fun definition() = ModuleDefinition {
    // Name must match the iOS view module so JS can require the same manager
    Name("ExpoRoomPlanView")

    // Register a View factory. The View itself will throw on Android.
    View(::RoomPlanView) {
      // Mirror props to satisfy the JS/TS surface. They are no-ops here.
      Prop("scanName") { _: RoomPlanView, _: String? -> }
      Prop("exportType") { _: RoomPlanView, _: String? -> }
      Prop("sendFileLoc") { _: RoomPlanView, _: Boolean? -> }
      Prop("running") { _: RoomPlanView, _: Boolean? -> }
      Prop("exportTrigger") { _: RoomPlanView, _: Double? -> }

      Events("onStatus", "onExported")
    }
  }
}
